lib/ansible/modules/cloud/azure/azure_rm_servicebus_facts.py

```diff
@@ -0,0 +1,479 @@
+
+options:
+    name:
+        description:
+            - Limit results to a specific servicebus.
+    resource_group:
+        description:
+            - Limit results in a specific resource group.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+    namespace:
+        description:
+            - Servicebus namespace name.
+            - A namespace is a scoping container for all messaging components.
+            - Multiple queues and topics can reside within a single namespace, and namespaces often serve as application containers.
+            - Required when C(type) is not C(namespace).
+    type:
+        description:
+            - Type of the resource.
+        choices:
+            - namespace
+            - queue
+            - topic
+            - subscription
+    topic:
+        description:
+            - Topic name.
+            - Required when C(type) is C(subscription).
+    show_sas_policies:
+        description:
+            - Whether to show the SAS policies.
+            - Not support when C(type) is C(subscription).
+            - Note if enable this option, the facts module will raise two more HTTP call for each resources, need more network overhead.
+        type: bool
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+
+'''
+
+EXAMPLES = '''
+- name: Get all namespaces under a resource group
+  azure_rm_servicebus_facts:
+    resource_group: foo
+    type: namespace
+
+- name: Get all topics under a namespace
+  azure_rm_servicebus_facts:
+    resource_group: foo
+    namespace: bar
+    type: topic
+
+- name: Get a single queue with SAS policies
+  azure_rm_servicebus_facts:
+    resource_group: foo
+    namespace: bar
+    type: queue
+    name: sbqueue
+    show_sas_policies: true
+
+- name: Get all subscriptions under a resource group
+  azure_rm_servicebus_facts:
+    resource_group: foo
+    namespace: bar
+    topic: sbtopic
+'''
+RETURN = '''
+servicebuses:
+    description: List of servicebus dicts.
+    returned: always
+    type: list
+    contains:
+        id:
+            description:
+                -  Resource Id
+            type: str
+        name:
+            description:
+                -  Resource name
+            type: str
+        location:
+            description:
+                -  The Geo-location where the resource lives.
+            type: str
+        namespace:
+            description:
+                - Namespace name of the queue or topic, subscription.
+            type: str
+        topic:
+            description:
+                - Topic name of a subscription.
+            type: str
+        tags:
+            description:
+                -  Resource tags.
+            type: str
+        sku:
+            description:
+                -  Porperties of Sku.
+            type: str
```

Comment:
what is that SKU exactly? i can't see sku parameter when resource is created, so perhaps we can omit it?

Reply:
This  tells the SKU of the namespace. Actually we cannot pass it  to the SDK when  we create or update the resource. But we should return the information user doesn't know in the facts module, right?

Reply:
is there a reason for user to know SKU values?
seems it can be selected in the portal, so not available in the api?

Reply:
no, not available.

I think the facts module should return as much as they can and in a user friendly way (why we add curated format)

Reply:
how about this api?
https://docs.microsoft.com/en-us/rest/api/servicebus/namespaces/createorupdate#sbsku

Reply:
added. Thanks